### ISSUE TITLE
🐛 BUG: Invalid markup generates a runtime error

### ISSUE THREAD

**Issue by godefroi** [withastro/compiler#353]
### What version of `astro` are you using?

0.25.4

### What package manager are you using?

yarn

### What operating system are you using?

Windows

### Describe the Bug

I am seeing an error attempting to render a very simple template. Support thread here: https://discord.com/channels/830184174198718474/958787462606495804

Minimal example linked below.

```
<!-- this works -->
{Object.keys(importedAuthors).map(author => <p>
	<div>hello</div>
</p>)}

<!-- this does not work -->
{Object.keys(importedAuthors).map(author => <p>
	<div>{author}</div>
</p>)}
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-rztbls?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
Nice sleuthing, @ryanfitzer! I took a stab at this one and I'm struggling to untangle where this is going wrong, but it looks like you're correct that mixing invalid HTML with expressions is the culprit. 

Somehow the trailing `)` in the expression is getting dropped before we generate the JS for the Astro component, so where we should be printing `)}` we're instead printing `}`. That's the cause of that error message.

I'll keep working to track down a fix!

**Comment by natemoo-re** [comment]
Related to https://github.com/withastro/compiler/issues/432

**Comment by MoustaphaDev** [comment]
I forgot to close this somehow, should've been closed by #353
